code/game/dna/genes/vg_powers.dm

```diff
@@ -242,6 +252,87 @@
 		for(var/mob/dead/observer/G in GLOB.player_list)
 			G.show_message("<i>Telepathic message from <b>[user]</b> ([ghost_follow_link(user, ghost=G)]) to <b>[target]</b> ([ghost_follow_link(target, ghost=G)]): [say]</i>")
 
+/obj/effect/proc_holder/spell/targeted/mindscan
+	name = "Scan Mind"
+	desc = "Offer people a chance to share their thoughts!"
+	charge_max = 0
+	clothes_req = 0
+	stat_allowed = 0
+	invocation_type = "none"
+	range = -2
+	selection_type = "range"
+	action_icon_state = "genetic_mindscan"
+	var/list/available_targets = list()
+
+/obj/effect/proc_holder/spell/targeted/mindscan/choose_targets(mob/user = usr)
+	var/list/targets = new /list()
+	var/list/validtargets = new /list()
+	var/turf/T = get_turf(user)
+	for(var/mob/living/M in range(14, T))
+		if(M && M.mind)
+			if(M == user)
+				continue
+			validtargets += M
+
+	if(!validtargets.len)
+		to_chat(user, "<span class='warning'>There are no valid targets!</span>")
+		start_recharge()
+		return
+
+	targets += input("Choose the target to listen to.", "Targeting") as null|mob in validtargets
+
+	if(!targets.len || !targets[1]) //doesn't waste the spell
+		revert_cast(user)
+		return
+
+	perform(targets, user = user)
+
+/obj/effect/proc_holder/spell/targeted/mindscan/cast(list/targets, mob/user = usr)
+	if(!ishuman(user))
+		return
+	for(var/mob/living/target in targets)
+		var/message = "You feel your mind expand briefly... (Click to send a message.)"
+		if(REMOTE_TALK in target.mutations)
+			message = "You feel [user.real_name] request a response from you... (Click here to project mind.)"
+		user.show_message("<span class='abductor'>You offer your mind to [target.name].</span>")
+		target.show_message("<span class='abductor'><A href='?src=[UID()];target=\ref[target];user=\ref[user]'>[message]</a></span>")
+		available_targets += target
+		addtimer(CALLBACK(src, .proc/removeAvailability, target), 100)
+
+/obj/effect/proc_holder/spell/targeted/mindscan/proc/removeAvailability(var/mob/living/target)
+	if(target in available_targets)
+		available_targets -= target
+
+/obj/effect/proc_holder/spell/targeted/mindscan/Topic(href, href_list)
+	var/mob/living/user
+	if(href_list["user"])
+		user = locate(href_list["user"])
```

Comment:
with the above change, use `locateUID` instead
(the reason is that as objects get deleted, their `\ref` values may be re-used, causing `locate` to return bogus objects. the UID system gets around this and ensures a locate only finds a single object, and not anything that happens to use the same `\ref` value)

Reply:
Interesting! That explains the slightly higher cost. Will do, and I'll keep that in mind for the future.